csrc/executor_kernel_arg.cpp

```diff
@@ -179,11 +179,12 @@ std::vector<std::byte> polymorphicValueToBytes(
     auto dtype_ = std::get<StructType>(dtype.type);
     auto struct_ = argument.as<Struct>();
     std::vector<std::byte> buffer;
-    for (const auto& field : dtype_.field_names) {
-      auto field_data = polymorphicValueToBytes(
-          struct_[field],
-          NVFUSER_MAYBE_STAR dtype_.types.at(field),
-          index_type);
+    for (const auto& field : dtype_.fields) {
+      if (!field.used_in_kernel) {
```

Comment:
Is this a fix? When is it `false`?

Reply:
This change does not fix anything. The previous behavior should be preserved. It can be false when we are defining structs different for lowering and eager evaluation. For example in tensor metadata, we have four fields `logical_size`, `logical_stride`, `alloc_size` and `alloc_stride`, but when lowered to kernel, we only use `logical_size` and `alloc_stride`, and this `used_in_kernel` is to implement this behavior.

Reply:
So, previously we were just sending all fields to device no matter used or not?

Reply:
No, previously, we had a `field_names` and a `types`, where `types` contains all of `logical_size`, `logical_stride`, `alloc_size` and `alloc_stride`, but `field_names` contains just `logical_size` and `alloc_stride`, which I believe is very confusing. That's why the refactor.